### PR TITLE
[ci/cd] Make the next branch easy to deploy

### PR DESCRIPTION
**Pull request by mdartic**

### CODE REVIEW

File: packages/designsystem/package.json [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-  "main": "src/index.ts",
+  "author": "Mathieu Dartigues",
+  "license": "MIT",
+  "main": "dist/index.cjs.js",
+  "module": "dist/index.es.js",
+  "types": "dist/src/index.d.ts",
   "scripts": {
     "lint": "eslint .",
-    "story:dev": "histoire dev",
+    "dev": "histoire dev",
```
Comment (Alketta) [inline packages/designsystem/package.json:12]: Why did you remove the 'story' before?

File: packages/designsystem/package.json [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
     "story:build": "histoire build",
-    "story:preview": "histoire preview"
+    "story:preview": "histoire preview",
+    "build": "rm -rf dist && vite build"
```
Comment (Alketta) [inline packages/designsystem/package.json:15]: Can we remove "story:build" to avoid confusion?

File: packages/designsystem/plugins/vee-validate.ts [hunk 3]
```diff
@@ -1,70 +0,0 @@
-// import { isAfter, isValid } from 'date-fns'
```
Comment (Alketta) [inline packages/designsystem/plugins/vee-validate.ts:1]: Can you explain why did you remove this file?

Comment (Alketta) [reply packages/designsystem/plugins/vee-validate.ts:1]: ah! Finally you've recreated it…

File: api/package.json [hunk 4]
```diff
@@ -74,13 +72,13 @@
...
     "feathers-authentication-management": "^5.0.0-pre.0",
     "feathers-hooks-common": "^7.0.3",
-    "feathers-locokit": "",
     "feathers-swagger": "^3.0.0",
     "knex": "^2.1.0",
     "knex-schema-inspector": "^3.0.1",
     "koa-helmet": "^6.1.0",
     "koa-ratelimit": "^5.0.1",
     "koa-static": "^5.0.0",
+    "marked": "^4.1.0",
```
Comment (Alketta) [inline api/package.json:81]: Are we still use it? If yes, maybe we can use the last version of it.

File: packages/nuxt-locokit/.gitignore [hunk 5]
```diff
@@ -34,7 +34,12 @@ coverage
 .nyc_output
 
 # VSCode
-.vscode
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
+!.vscode/*.code-snippets
```
Comment (Alketta) [inline packages/nuxt-locokit/.gitignore:42]: I still don't think these files should be committed.

File: packages/nuxt-locokit/src/module.ts [hunk 6]
```diff
@@ -442,34 +397,100 @@ export default defineNuxtModule<ModuleOptions>({
       /**
        * Register auth pages
        */
+      console.log('auth enabled ?', submodules.auth.enabled)
```
Comment (Alketta) [inline packages/nuxt-locokit/src/module.ts:400]: An another log?

File: packages/nuxt-locokit/src/module.ts [hunk 7]
```diff
@@ -442,34 +397,100 @@ export default defineNuxtModule<ModuleOptions>({
...
       if (submodules.auth.enabled) {
+        console.log('[nuxt-locokit] registering auth routes...')
         const prefix = submodules.auth.prefix
-        pages.push(...getAuthPages(prefix, pagesDir))
+        pages.push(
```
Comment (Alketta) [inline packages/nuxt-locokit/src/module.ts:404]: Why this changed?

Comment (Alketta) [reply packages/nuxt-locokit/src/module.ts:404]: Why keep `packages/nuxt-locokit/src/runtime/routes.ts`, then?

File: packages/nuxt-locokit/src/runtime/stores/auth.ts [hunk 8]
```diff
@@ -6,7 +6,9 @@ import { ref, useCookie } from '#imports'
 export const useStoreAuth = defineStore('auth', () => {
   const loading = ref(false)
   const isAuthenticated = ref(false)
-  const error = ref<Error | null>(null)
+  const error = ref<
+    Error | null | { code: string | null; content: string | null }
```
Comment (Alketta) [inline packages/nuxt-locokit/src/runtime/stores/auth.ts:10]: What wlll contains "content"?

File: packages/nuxt-locokit/src/runtime/pages/auth/login.vue [hunk 9]
```diff
@@ -0,0 +1,83 @@
+<template>
```
Comment (Alketta) [inline packages/nuxt-locokit/src/runtime/pages/auth/login.vue:1]: Why this new file while SignIn already exists?

File: packages/nuxt-locokit/src/module.ts [hunk 10]
```diff
@@ -149,29 +138,6 @@ export default defineNuxtModule<ModuleOptions>({
     //   })
     // }
 
-    /**
-     * Register all layouts
-     */
-    addLayout(
-      {
-        src: resolve(layoutsDir, './WithHeader.vue'),
```
Comment (Alketta) [inline packages/nuxt-locokit/src/module.ts:157]: I didn't see change in file using layout. Is this normal?

File: packages/designsystem/package.json (hunk 1 above)
Comment (mdartic) [reply packages/designsystem/package.json:12]: This is for convenience, by running `turbo run dev`, all scripts `dev` are runned simultaneously.
So the "histoire"book. I think it's useful ? WDYT ?

File: packages/designsystem/package.json (hunk 2 above)
Comment (mdartic) [reply packages/designsystem/package.json:15]: Mmm, this is not the same goal ?
`story:build` build the "histoire"book,
`build` build the design system package.

I prefer keep them both ?

File: packages/designsystem/plugins/vee-validate.ts (hunk 3 above)
Comment (mdartic) [reply packages/designsystem/plugins/vee-validate.ts:1]: In fact I move the file to be under `src/`. This is for typescript, as the rootDir is `src/`. (check the `tsconfig.json`)

File: api/package.json (hunk 4 above)
Comment (mdartic) [reply api/package.json:81]: Yup, we are using it for mailing purpose, in the authmanagement system.
I check the latest version, `7.0.2` ! We are late.
I open a new issue for this, as it is not related to build.

see https://github.com/locokit/locokit/issues/247

File: packages/nuxt-locokit/.gitignore (hunk 5 above)
Comment (mdartic) [reply packages/nuxt-locokit/.gitignore:42]: Well, that's `nuxt` who create this file.
So, maybe they need to ? Or at least, it could be useful for other devs ? (don't really know)

File: packages/nuxt-locokit/src/module.ts (hunk 7 above)
Comment (mdartic) [reply packages/nuxt-locokit/src/module.ts:404]: Mmm, this has changed because I encounter (a lot) issues when building the nuxt module.
I discover that some paths are not really working, layouts are causing issues, etc.
I ended up to push all pages in the module files, but you're right, I remove the `routes.ts` file.

File: packages/nuxt-locokit/src/module.ts (hunk 6 above)
Comment (mdartic) [reply packages/nuxt-locokit/src/module.ts:400]: Removed

File: packages/nuxt-locokit/src/runtime/pages/auth/login.vue (hunk 9 above)
Comment (mdartic) [reply packages/nuxt-locokit/src/runtime/pages/auth/login.vue:1]: Mistake from me... Removed

File: packages/nuxt-locokit/src/module.ts (hunk 10 above)
Comment (mdartic) [reply packages/nuxt-locokit/src/module.ts:157]: Yes.
Adding layout was clearly breaking the build and usage of the nuxt module, as it was referencing files in the module.
This seems not OK, but I didn't find any documentation on this.

Pages using layouts inside the module are OK.
Pages in the app using our `nuxt-locokit` module won't have access to our layouts, until our layouts need other files of the module (like paths).

So, for now, I don't have other work around for this. I simply remove the `addLayout`.
And this is fine as we were not using layouts in our nuxt playground / app.

File: packages/nuxt-locokit/src/runtime/stores/auth.ts (hunk 8 above)
Comment (mdartic) [reply packages/nuxt-locokit/src/runtime/stores/auth.ts:10]: An explanation of the error ?
Don't remember why I did this, but I think this is related to Feathers Errors.
I can remove it as, after checking the code, I can't see any reference to a `error.content`... WDYT ?

File: lck-api.env [hunk 11]
```diff
@@ -0,0 +1,65 @@
+LCK_PORT=8002
```
[the remark this answers is not in the document]
Comment (mdartic) [reply lck-api.env:1]: You're right, this is not the right place.
This file is here for docker help, when you'll create a locokit instance on your own.
Maybe we can add it to the documentation, as a file to download, or create another folder in our source code.
I don't know what's the best way.
Any ideas ?
This is the same for `docker-compose.[prod, rpi, starter].yml`. These files are only for deployment / discovery.
Not for development.

[the remark this answers is not in the document]
Comment (Alketta) [reply lck-api.env:1]: We already have this file, haven’t? with https://github.com/locokit/locokit/blob/next/api/.env.example
Some var are different but there are similar. Can we name it like the docker-compose: `env.[test,…]`?

File: packages/designsystem/package.json (hunk 2 above)
Comment (Alketta) [reply packages/designsystem/package.json:15]: ok

File: packages/designsystem/package.json (hunk 1 above)
Comment (mdartic) [reply packages/designsystem/package.json:12]: That's right, in this case you don't need to run `turbo run dev`. 
You could just individual npm scripts, just like before.

File: packages/nuxt-locokit/.gitignore (hunk 5 above)
Comment (Alketta) [reply packages/nuxt-locokit/.gitignore:42]: hmmm, I have doubt. I don't want to block this MR with this.

File: packages/nuxt-locokit/src/runtime/stores/auth.ts (hunk 8 above)
Comment (Alketta) [reply packages/nuxt-locokit/src/runtime/stores/auth.ts:10]: Yep, same as "code" (for now :smile: ) and I am not sure if we will use it.

File: packages/designsystem/plugins/vee-validate.ts (hunk 3 above)
Comment (Alketta) [reply packages/designsystem/plugins/vee-validate.ts:1]: For Git it is a deletion and a creation not a move.

File: lck-api.env (hunk 11 above)
[the remark this answers is not in the document]
Comment (mdartic) [reply lck-api.env]: Updated like this :
* move `lck-api.env` to `api/.env.prod.example`
* remove `docker-compose.starter.yml`
* add explanations in `prod` and `rpi` docker-compose files

**Review by Alketta — APPROVED** [review]
